src/test/java/net/sf/jabref/gui/cleanup/CleanupActionsListModelTest.java

```diff
@@ -0,0 +1,90 @@
+import net.sf.jabref.logic.formatter.bibtexfields.EraseFormatter;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import javax.swing.event.ListDataEvent;
+import javax.swing.event.ListDataListener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class CleanupActionsListModelTest {
+
+    @Test
+    public void resetFiresItemsChanged() throws Exception {
+        CleanupActionsListModel model = new CleanupActionsListModel(Collections.emptyList());
+        ListDataListener listener = mock(ListDataListener.class);
+        model.addListDataListener(listener);
+        FieldFormatterCleanups defaultFormatters = mock(FieldFormatterCleanups.class);
+
+        model.reset(defaultFormatters);
+
+        ArgumentCaptor<ListDataEvent> argument = ArgumentCaptor.forClass(ListDataEvent.class);
+        verify(listener).contentsChanged(argument.capture());
+        assertEquals(ListDataEvent.CONTENTS_CHANGED, argument.getValue().getType());
+    }
+
+    @Test
+    public void resetSetsFormattersToPassedList() throws Exception {
+        CleanupActionsListModel model = new CleanupActionsListModel(Collections.emptyList());
+        FieldFormatterCleanups defaultFormatters = mock(FieldFormatterCleanups.class);
+        List<FieldFormatterCleanup> formatters = Arrays.asList(new FieldFormatterCleanup("test", new EraseFormatter()));
+        when(defaultFormatters.getConfiguredActions()).thenReturn(formatters);
+
+        model.reset(defaultFormatters);
+
+        assertEquals(formatters, model.getAllActions());
+    }
+
+    public List<FieldFormatterCleanup> getDefaultFieldFormatterCleanups() {
+        FieldFormatterCleanups formatters = FieldFormatterCleanups.DEFAULT_SAVE_ACTIONS;
+        //new ArrayList because configured actions is an unmodifiable collection
+        return new ArrayList<>(formatters.getConfiguredActions());
+    }
+
+    @Test
+    public void removedAtIndexOkay() {
+
+        CleanupActionsListModel model = new CleanupActionsListModel(getDefaultFieldFormatterCleanups());
+        ListDataListener listener = mock(ListDataListener.class);
+        model.addListDataListener(listener);
+        model.removeAtIndex(0);
+
+        ArgumentCaptor<ListDataEvent> argument = ArgumentCaptor.forClass(ListDataEvent.class);
+        verify(listener).intervalRemoved(argument.capture());
+        assertEquals(ListDataEvent.INTERVAL_REMOVED, argument.getValue().getType());
+
+    }
+
+    @Test
+    public void removedAtIndexMinus1() {
+
+        CleanupActionsListModel model = new CleanupActionsListModel(getDefaultFieldFormatterCleanups());
+        ListDataListener listener = mock(ListDataListener.class);
+        model.addListDataListener(listener);
```

Comment:
I am missing an assertion here and in the next test. Probably you want to check that the model still contains the same items and use `verifyZeroInteractions(mockObject)` to verify that no change event was raised. Please also change the test name to reflect the expected outcome, i.e. removedAtIndexMinus1DoesNothing